### PR TITLE
Sam/add support for range in stock chart

### PR DESCRIPTION
**Pull request by urica**

### CODE REVIEW

File: highcharts/src/main/java/us/ascendtech/highcharts/client/ChartOptions.java [hunk 1]
```diff
@@ -417,4 +420,15 @@ public final ChartOptions setzAxis(ZAxis zAxis) {
 		this.zAxis = zAxis;
 		return this;
 	}
+
+	@JsOverlay
+	public final RangeSelector getRangeSelector() {
```
Comment (bpetridean) [inline highcharts/src/main/java/us/ascendtech/highcharts/client/ChartOptions.java]: @JsProperty is better suited here
I mean as long as the prop exists in js world you do not need to create it in java world, just provide the means to access it, hence the @JsProperty

File: highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+ * @author Samuel Palacean
+ * Docs: https://api.highcharts.com/highstock/rangeSelector
+ */
+@JsType(isNative = true, name = "Object", namespace = JsPackage.GLOBAL)
+public class RangeSelector {
+    private Boolean allButtonsEnabled;
+    private double buttonSpacing;
+    private String buttonTheme;
+    private String dropdown;
+    private Boolean enabled;
```
Comment (bpetridean) [inline highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java:21]: why not primitive type `boolean`?
Are you also handling null values?

File: highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    private Boolean floating;
```
Comment (bpetridean) [inline highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java:22]: same as above

File: highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/Buttons.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+/**
+ * @author Samuel Palacean
+ * Docs: https://api.highcharts.com/highstock/rangeSelector.buttons
+ */
+@JsType(isNative = true, name = "Object", namespace = JsPackage.GLOBAL)
+public class Buttons {
+    private double count;
+    private double offsetMax;
+    private double offsetMin;
+    private Boolean preserveDataGrouping;
```
Comment (bpetridean) [inline highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/Buttons.java:19]: Why not primitive type `boolean`?
Are you also handling null values?

**Review by bpetridean — CHANGES_REQUESTED** [review]

File: highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java (hunk 2 above)
Comment (urica) [reply highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java:21]: image


based on documentation there is also undefined value possible. That`s why Boolean.

File: highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java (hunk 3 above)
Comment (urica) [reply highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java:22]: same as above

File: highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/Buttons.java (hunk 4 above)
Comment (urica) [reply highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/Buttons.java:19]: same as above

File: highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java (hunk 2 above)
Comment (bpetridean) [reply highcharts/src/main/java/us/ascendtech/highcharts/client/chartoptions/rangeselector/RangeSelector.java:21]: got it 👍🏼

File: highcharts/src/main/java/us/ascendtech/highcharts/client/ChartOptions.java (hunk 1 above)
Comment (urica) [reply highcharts/src/main/java/us/ascendtech/highcharts/client/ChartOptions.java]: done

**Review by bpetridean — APPROVED** [review]
🎱

**Pull request merged**